src/ExecutorContainer.ts

```diff
@@ -4,8 +4,19 @@ import NodeJSExecutor from "./executors/NodeJSExecutor";
 import NonInteractiveCodeExecutor from "./executors/NonInteractiveCodeExecutor";
 import PrologExecutor from "./executors/PrologExecutor";
 import PythonExecutor from "./executors/python/PythonExecutor";
+import CppExecutor from './executors/CppExecutor';
 import ExecuteCodePlugin, {LanguageId} from "./main";
 
+const interactiveExecutors: Partial<Record<LanguageId, any>> = {

```

Comment:
I really like this change. This is a much more extensible and maintainable approach than what I did. Thanks!